packages/xforms-engine/src/error/RankError.ts

```diff
@@ -0,0 +1,20 @@
+import { XFormsSpecViolationError } from './XFormsSpecViolationError.ts';
```

Comment:
This suggestion is to keep Rank's error definitions together instead of individual files since each class is straightforward.

Reply:
Ah I see you addressed this before me. We've generally settled on a convention of 1 _exported class_ to 1 same-named module. I think this is the most common idiom in most TS/JS projects (probably inherited from Java). I don't think we've applied the convention as consistently as maybe we ought to.

But I think **especially** for things like errors it would be valuable, because the _eventual intent_ is that they'll be used to produce documentation of which errors can occur. (The value will probably be most visible in generated documentation, where each class/module is a unit.)

Reply:
I've separated the errors to different files